File: packages/web/src/types/recipe.ts
import { GroceryItemUnit } from '../enums/groceryItem'
import { MealType } from '@kairos/shared'
import { RecipeDishType } from '@kairos/shared'

export interface IRecipeIngredient {
    name: string
    quantity: number
    unit: GroceryItemUnit
}

export interface IRecipe {
    id: string
    projectId: string
    name: string
    imagePath?: string
    externalLink?: string
    ingredients: IRecipeIngredient[]
    instructions?: string[]
    mealTypes?: MealType[]
    dishTypes?: RecipeDishType[]
    createdAt: string
    updatedAt: string
    visibility?: "private"
    ownerId?: string
}

File: packages/ios/Kairos/Kairos/Models/Recipe.swift
import Foundation

/// Web-app type (not in shared types). See `models.md`.
struct Recipe: Codable, Identifiable, Sendable, Equatable {
    let id: String
    let projectId: String
    var name: String
    var imagePath: String?
    var externalLink: String?
    var ingredients: [RecipeIngredient]
    var instructions: [String]?
    var mealTypes: [MealType]?
    var dishTypes: [RecipeDishType]?
    let createdAt: String
    var updatedAt: String
    var visibility: String?
    var ownerId: String?
}

struct RecipeIngredient: Codable, Sendable, Equatable {
    var name: String
    var quantity: Double
    var unit: GroceryItemUnit
}
